### ISSUE TITLE
Non-agent tokens shouldn't attend to agent tokens if self.mode == "wm_agent"

### ISSUE THREAD

**Issue by jw9730** [nicklashansen/dreamer4#5]
Dear Nicklas, thank you very much for this implementation! It's been quite helpful for understanding Dreamer 4.

I'm writing to ask about one of the attention patterns in `dreamer4/model.py` in lines 230-232:

```
        if self.mode == "wm_agent":
            # full mixing across modalities
            return torch.ones((S, S), dtype=torch.bool, device=device)
```

This allows non-agent tokens to attend to everything, but based on what I understand from Section 3.3 of the [original Dreamer 4 paper](https://arxiv.org/abs/2509.24527) and the [referenced JAX implementation](https://github.com/edwhu/dreamer4-jax/blob/main/dreamer/models.py), it seems non-agent tokens should not attend to agent tokens to prevent task conditioning from leaking into frame prediction.

If this observation is correct, the implementation could be something like:

```
        if self.mode == "wm_agent":
            # non-agent tokens: can attend to everything EXCEPT agent tokens
            # agent tokens: can attend to everything
            is_q_agent = (q_mod == int(Modality.AGENT))
            is_k_agent = (k_mod == int(Modality.AGENT))
            allow = torch.ones((S, S), dtype=torch.bool, device=device)

            # non-agent queries cannot see agent keys
            allow_non_agent_q = ~is_q_agent
            allow = torch.where(allow_non_agent_q, ~is_k_agent, allow)
            return allow
```

Could you verify whether the current choice is intended? Thanks in advance.

**Comment by witherhoard99** [comment]
I'm not the author, but I think it's because wm_agent is currently just a placeholder. Currently, there is no BC/RL training as the dynamics model currently deals only with diffusion shortcut forcing (no actions or reward/policy heads). The default during pretraining is wm_agent_isolated, so wm_agent isn't used unless set manually.

**Comment by jw9730** [comment]
That makes sense, thank you for the clarification!

**Comment by nicklashansen** [comment]
Correct! Perhaps it would be better to remove this logic altogether

**Comment by Arison591** [comment]
I can prepare a small PR that removes the unsupported wm_agent placeholder path and its exposed configuration, while keeping the implemented wm_agent_isolated behavior covered by tests. I will first trace all mode references so the change does not leave a selectable but partially implemented mode.
